frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { RequestSizeInput, withHandlePromise } from '@console/internal/components/utils/index';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import {
+  createModalLauncher,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalTitle,
+} from '@console/internal/components/factory';
+import { k8sPatch, K8sResourceKind } from '@console/internal/module/k8s';
+import { OCSServiceModel } from '../../../models';
+import './_add-capacity-modal.scss';
+
+export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps) => {
+  const { ocsConfig, close, cancel } = props;
+
+  const requestSizeUnit = 'Ti';
+  const [requestSizeValue, setRequestSizeValue] = React.useState('');
+  const [storageClass, setStorageClass] = React.useState('');
+  const [inProgress, setProgress] = React.useState(false);
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    setProgress(true);
+    const presentCapacity = ocsConfig.spec.storageDeviceSets[0].count;
+    const newObj = _.cloneDeep(ocsConfig.spec);
+    newObj.storageDeviceSets[0].count =
+      parseInt(presentCapacity, 10) + parseInt(requestSizeValue, 10) * 3;
+    const patch = {
+      op: 'replace',
+      path: '/spec',
+      value: newObj,
+    };
+    props
+      .handlePromise(k8sPatch(OCSServiceModel, ocsConfig, [patch]))
+      .then(() => {
+        setProgress(false);
+        close();
+      })
+      .catch((error) => {
+        cancel();
+        setProgress(false);
+        throw error;
+      });
+  };
+
+  const handleRequestSizeInputChange = (capacityObj: any) => {
+    setRequestSizeValue(capacityObj.value);
+  };
+
+  const handleStorageClass = (sc: K8sResourceKind) => {
+    setStorageClass(sc.metadata.name);
+  };
+
+  return (
+    <form onSubmit={submit} className="modal-content add-capacity-modal__class">
+      <ModalTitle>Add Capacity</ModalTitle>
+      <ModalBody>
+        Increase the capacity of <strong>{ocsConfig.metadata.name}</strong>.
+        <div className="add-capacity-modal__padding">
+          <div className="form-group">
+            <label className="control-label" htmlFor="request-size-input">
+              Requested Capacity
+              <span className="add-capacity-modal__used">
+                Used:{' '}
+                {
+                  ocsConfig.spec.storageDeviceSets[0].dataPVCTemplate.spec.resources.requests
+                    .storage
+                }
+              </span>
+            </label>
+            <RequestSizeInput
+              name="requestSize"
+              placeholder={requestSizeValue}
+              onChange={handleRequestSizeInputChange}
+              defaultRequestSizeUnit={requestSizeUnit}
+              defaultRequestSizeValue={requestSizeValue}
+              dropdownUnits={{}}
```

Comment:
Since there is only one unit, it looks like this:- 
![Screenshot from 2019-08-29 19-00-48](https://user-images.githubusercontent.com/12200504/63944559-5ebba280-ca8f-11e9-84e5-4f6e56e75964.png)

Reply:
ack